Sources/HipMobileUI/Pages/UserRatingPage.xaml

```diff
@@ -17,163 +17,169 @@
              x:Class="PaderbornUniversity.SILab.Hip.Mobile.UI.Pages.UserRatingPage"
-             xmlns:controls1="clr-namespace:PaderbornUniversity.SILab.Hip.Mobile.UI.Controls;assembly=HipMobileUI"
-             xmlns:behaviors="clr-namespace:PaderbornUniversity.SILab.Hip.Mobile.UI.Behaviors;assembly=HipMobileUI"
              xmlns:helpers="clr-namespace:PaderbornUniversity.SILab.Hip.Mobile.UI.Helpers;assembly=HipMobileUI"
-             xmlns:views="clr-namespace:PaderbornUniversity.SILab.Hip.Mobile.UI.Views;assembly=HipMobileUI"
              Title="{Binding Headline}">
     <ContentPage.Content>
-        <AbsoluteLayout>
-            <Image Source="{Binding Image}" Margin="5" AbsoluteLayout.LayoutBounds="0,0,1,0.6" AbsoluteLayout.LayoutFlags="All" Aspect="AspectFill" Opacity="0.08"/>
-            <Grid AbsoluteLayout.LayoutBounds="1,1,1,1" AbsoluteLayout.LayoutFlags="All" RowSpacing="0">
-                <Grid.ColumnDefinitions>
-                    <ColumnDefinition Width="Auto" />
-                </Grid.ColumnDefinitions>
-                <Grid.RowDefinitions>
-                    <RowDefinition Height="0.3*" />
-                    <RowDefinition Height="0.3*" />
-                    <RowDefinition Height="0.3*" />
-                    <RowDefinition Height="0.1*" />
-                </Grid.RowDefinitions>
-
-                <StackLayout Grid.Row="0" Orientation="Vertical" Spacing="0" VerticalOptions="Center">
-                    <Label Text="{helpers:Translate UserRating_Average_Rating}" FontAttributes="Bold" FontSize="17" VerticalOptions="Center" HorizontalOptions="Center" Margin="0,0,0,15" />
-                    <Label Text="{Binding RatingAverage}" FontAttributes="Bold" FontSize="24" VerticalOptions="Center" HorizontalOptions="Center" />
-                    <StackLayout Orientation="Horizontal" HorizontalOptions="Center" VerticalOptions="Center" Spacing="8">
-                        <Image WidthRequest="25" HeightRequest="25" Source="{Binding Star1}" VerticalOptions="Center"/>
-                        <Image WidthRequest="25" HeightRequest="25" Source="{Binding Star2}" VerticalOptions="Center"/>
-                        <Image WidthRequest="25" HeightRequest="25" Source="{Binding Star3}" VerticalOptions="Center"/>
-                        <Image WidthRequest="25" HeightRequest="25" Source="{Binding Star4}" VerticalOptions="Center"/>
-                        <Image WidthRequest="25" HeightRequest="25" Source="{Binding Star5}" VerticalOptions="Center"/>
+        <AbsoluteLayout x:Name="OuterAbsoluteLayout">
+            <!-- The image layout definition is set up in code -->
+            <Image x:Name="Image" Source="{Binding Image}" Margin="5" Aspect="AspectFill" Opacity="0.08"/>
+            <!-- The row and column definition is set up in code -->
+            <Grid x:Name="OuterGrid" AbsoluteLayout.LayoutBounds="1,1,1,1" AbsoluteLayout.LayoutFlags="All" RowSpacing="0" ColumnSpacing="0">
```

Comment:
This is quite a huge amount of `Grids`. Are they all required? Did you try using less and maybe `SetColumnSpan()` or `SetRowSpan()` for elements stretching across multiple columns/rows?

Reply:
Yes, I need all the three grids, because the screen in landscape contains of two sides (which is why I need the OuterGrid) and each side contains 2 elements (which is why I need the other 2 grids).

Reply:
Ok, it's working so I won't go more in detail here.